### PR TITLE
Adding parallel training functionality to the Experiment and Model

### PR DESCRIPTION
**Pull request by hamlinliu17**
Resolving the issue [here](https://github.com/vc1492a/tidd/issues/84) finally. We can now specify the cuda devices specifically the indices on cuda devices.

### CODE REVIEW

File: tidd/modeling.py [hunk 1]
```diff
@@ -95,11 +101,28 @@ def fit(self, max_epochs: int, verbose: bool = False, callbacks: list = None) ->
             self.callbacks.append(ShowGraphCallback())
 
         # train the model
+<<<<<<< HEAD
```
Comment (vc1492a) [inline tidd/modeling.py:104]: Please remove the inserts in the file from a merge.

File: tidd/modeling.py [hunk 2]
```diff
@@ -95,11 +101,28 @@ def fit(self, max_epochs: int, verbose: bool = False, callbacks: list = None) ->
...
+                    lr=self.learning_rate,
+                    cbs=callbacks
+                )
+        else:
+            self.learner.fit(
+                max_epochs,
+                lr=self.learning_rate,
+                cbs=callbacks
+            )
+=======
```
Comment (vc1492a) [inline tidd/modeling.py:119]: Another one

File: tidd/modeling.py [hunk 3]
```diff
@@ -95,11 +101,28 @@ def fit(self, max_epochs: int, verbose: bool = False, callbacks: list = None) ->
...
         self.learner.fit(
             max_epochs,
             lr=self.learning_rate,
             cbs=self.callbacks
         )
+>>>>>>> dev
```
Comment (vc1492a) [inline tidd/modeling.py:125]: one more

**Review by vc1492a — CHANGES_REQUESTED** [review]
@hamlinliu17 please clean up the merge artifacts and then commit the changes. Once you see that the tests pass in [Travis CI](https://travis-ci.com/github/vc1492a/tidd/branches), please go ahead and merge into the `dev` branch. 

Following that, please do the following: 

- Increment the version number and update the version number throughout the code (`setup.py`, `readme.md`, etc.). 
- Update `changelog.md` to reflect the new version number and describe the changes made. 
- Create a distribution in `dist` directory once you have run the tests and verified that the code works as intended. 
- Commit these changes to the `dev` branch and then ping me to review. We'll then pull the changes into the `main` branch. Thanks!

**Comment by hamlinliu17** [comment]
@vc1492a I am going to merge this but the coverage of the tests has been lowered now.

**Pull request merged**